File: frontend/src/services/company.ts
import { apiClient } from "../utils/apiClient";

const API = "/api/v1/companies";

// ------------------------------------------------------------------ //
// Types                                                               //
// ------------------------------------------------------------------ //

export interface CompanyData {
    id: number;
    uuid: string;
    name: string;
    display_name: string;
    status: string;
    license_type: string | null;
    max_sites: number;
    max_agents: number;
    max_users: number;
    primary_contact: string | null;
    contact_email: string | null;
    contact_phone: string | null;
    timezone: string | null;
    logo_url: string | null;
    theme: string | null;
    notes: string | null;
    enabled: boolean;
    is_global: boolean;
    site_count: number;
    agent_count: number;
    integration_count: number;
    created_at: string | null;
    updated_at: string | null;
}

export interface CompanyListData {
    count: number;
    items: CompanyData[];
}

export interface CompanyCreateInput {
    name: string;
    display_name: string;
    status?: string;
    license_type?: string;
    max_sites?: number;
    max_agents?: number;
    max_users?: number;
    primary_contact?: string;
    contact_email?: string;
    contact_phone?: string;
    timezone?: string;
    logo_url?: string;
    theme?: string;
    notes?: string;
    enabled?: boolean;
}

export interface CompanyUpdateInput {
    name?: string;
    display_name?: string;
    status?: string;
    license_type?: string;
    max_sites?: number;
    max_agents?: number;
    max_users?: number;
    primary_contact?: string;
    contact_email?: string;
    contact_phone?: string;
    timezone?: string;
    logo_url?: string;
    theme?: string;
    notes?: string;
    enabled?: boolean;
}

// ------------------------------------------------------------------ //
// API                                                                 //
// ------------------------------------------------------------------ //

export const companiesApi = {
    async list(): Promise<CompanyData[]> {
        return apiClient<CompanyData[]>(API);
    },

    async listSummary(): Promise<CompanyData[]> {
        return apiClient<CompanyData[]>(`${API}/summary`);
    },

    async get(id: number): Promise<CompanyData> {
        return apiClient<CompanyData>(`${API}/${id}`);
    },

    async create(data: CompanyCreateInput): Promise<CompanyData> {
        return apiClient<CompanyData>(API, {
            method: "POST",
            json: data,
        });
    },

    async update(id: number, data: CompanyUpdateInput): Promise<CompanyData> {
        return apiClient<CompanyData>(`${API}/${id}`, {
            method: "PUT",
            json: data,
        });
    },

    async remove(id: number): Promise<void> {
        return apiClient<void>(`${API}/${id}`, {
            method: "DELETE",
        });
    },
};

File: backend/app/schemas/company.py
"""
Mission Control Company Schemas

Pydantic request/response models for company management.
Sprint 2.9 - Multi-Tenant & Multi-Site Platform.
"""

from datetime import datetime

from pydantic import BaseModel, Field


class CompanyCreate(BaseModel):
    """Request body for creating a company."""

    name: str = Field(..., min_length=1, max_length=200)
    display_name: str = Field(..., min_length=1, max_length=200)
    status: str = Field("active", max_length=20)
    license_type: str | None = None
    max_sites: int = 10
    max_agents: int = 100
    max_users: int = 50
    primary_contact: str | None = None
    contact_email: str | None = None
    contact_phone: str | None = None
    timezone: str | None = None
    logo_url: str | None = None
    theme: str | None = None
    notes: str | None = None
    enabled: bool = True


class CompanyUpdate(BaseModel):
    """Request body for updating a company."""

    name: str | None = Field(None, min_length=1, max_length=200)
    display_name: str | None = Field(None, min_length=1, max_length=200)
    status: str | None = None
    license_type: str | None = None
    max_sites: int | None = None
    max_agents: int | None = None
    max_users: int | None = None
    primary_contact: str | None = None
    contact_email: str | None = None
    contact_phone: str | None = None
    timezone: str | None = None
    logo_url: str | None = None
    theme: str | None = None
    notes: str | None = None
    enabled: bool | None = None


class CompanyResponse(BaseModel):
    """Response for a company."""

    id: int
    uuid: str
    name: str
    display_name: str
    status: str
    license_type: str | None = None
    max_sites: int
    max_agents: int
    max_users: int
    primary_contact: str | None = None
    contact_email: str | None = None
    contact_phone: str | None = None
    timezone: str | None = None
    logo_url: str | None = None
    theme: str | None = None
    notes: str | None = None
    enabled: bool
    is_global: bool

    site_count: int = 0
    agent_count: int = 0
    integration_count: int = 0

    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class CompanyListResponse(BaseModel):
    """Response for listing companies."""

    count: int
    items: list[CompanyResponse]


class CompanySummaryResponse(BaseModel):
    """Compact response for company selector dropdown."""

    id: int
    uuid: str
    name: str
    display_name: str
    status: str
    enabled: bool

    model_config = {"from_attributes": True}
